src/platform/gl/sdl_gl_platform.cpp

```diff
@@ -0,0 +1,84 @@
+#include "platform/platform.h"
+#include "common/error.h"
+
+#include "SDL.h"
+#include "GL/gl3w.h"
+
+struct gl_device
+{
+    SDL_GLContext ctx;
+};
+
+namespace vx
+{
+void platform_init(vx::platform* platform, const char* title, int2 initial_size)
+{
+    SDL_Window* sdl_window;
+    SDL_GLContext sdl_gl_context;
+
+    SDL_Init(SDL_INIT_VIDEO);
+
+    sdl_window = SDL_CreateWindow(
+        title,
+        SDL_WINDOWPOS_CENTERED,
+        SDL_WINDOWPOS_CENTERED,
+        initial_size.x,
+        initial_size.y,
+        SDL_WINDOW_OPENGL | SDL_WINDOW_RESIZABLE);
+
+    if (!sdl_window)
+        vx::fatal("SDL_CreateWindow failed with error: %s", SDL_GetError());
+
+    sdl_gl_context = SDL_GL_CreateContext(sdl_window);
+
+    if (gl3wInit() == -1)
+        vx::fatal("gl3wInit failed");
+
+    platform->window = (vx::window*)sdl_window;
+    platform->gpu = (vx::gpu_device*)sdl_gl_context;
+}
+
+void platform_quit(vx::platform* platform)
```

Comment:
since we're defining these within the `vx` namespace, there's no need to include the namespace name here and elsewhere in this file